### ISSUE TITLE
false no-unnecessary-callback-wrapper error

### ISSUE THREAD

**Issue by kevinphelps** [palantir/tslint#3456]
### Bug Report

- __TSLint version__: 5.8.0
- __TypeScript version__: 2.6.1
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
export function execute(command: string, environment: { [key: string]: string } = { }) {
  const spawnOptions: SpawnOptions = {
    stdio: 'inherit',
    shell: true,
    env: makeChildProcessEnv(environment)
  };

  return promisifyProcess(command, () => spawn(command, [], spawnOptions));
}

export function executeParallel(...commands: string[]) {
  return Promise.all(commands.map(command => execute(command))); // false no-unnecessary-callback-wrapper error
}
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "no-unnecessary-callback-wrapper": true
  }
}
```

#### Actual behavior
False `no-unnecessary-callback-wrapper` error. The signature of `execute` does not match the parameter of `Array.prototype.map` resulting in a type error if you just use `commands.map(execute)` as the `no-unnecessary-callback-wrapper` rule suggests.

#### Expected behavior
No error.

**Comment by ajafff** [comment]
duplicate of #2430
